lib/sycamore/sycamore/utils/pdf.py

```diff
@@ -234,3 +235,24 @@ def read_stderr(fh, q):
 
         t_out.join()
         t_err.join()
+
+
+@requires_modules(["pdfminer", "pdfminer.utils"], extra="local-inference")
+def pdf_to_pages(file_name: str) -> Generator[Any, None, None]:
```

Comment:
Might suggest putting this in pdf_miner.py. Seems pretty tied to that as opposed to generally useful for PDFs.